Populate README

hershen opened the pull request:
Basic README.

hershen committed:
Populate README

Add information on requirements, how to compile, and how to run the
hershensh shell.

oorijsonar commented on README.md:
```diff
@@ -1 +1,33 @@
+
+## Test suite
+
+From the `build` directory run `./runTests`.
```
1. I think that this section should come before the usage section as you test before you run the binary.
2. Set up cmake so that we have a test target then you will run the tests with make test

hershen replied:
1. I think everyone will be running the binary, but not everyone needs to run the tests. So I actually think this order makes more sense.
2. It was already set up, so I guess there are 2 ways to run the tests. I'll update the readme.

oorijsonar replied:
1. I assumed that the readme was meant for developers. If it's not then you are right.

hershen replied:
1. It's meant for everyone (including developers).

oorijsonar commented on README.md:
```diff
@@ -1 +1,33 @@
+
+This will create the `hershensh` binary.
+
+## Usage
```
You don't have to run it from the build directory (I hope!).

hershen replied:
Clarified it.

hershen committed:
Improvements to README.md

1. Rename hershensh -> hersh.
2. Remove Catch2 requirement.
3. Clarification how to run the hersh binary.
4. Clarification how to run the test suite.

oorijsonar commented on README.md:
```diff
@@ -1 +1,33 @@
+This will create the `hersh` binary inside the `build` directory.
+
+## Usage
+Run the `hersh` binary located in the `build` directory to open the hersh shell.
```
I think I don't get what this is aimed for. Will you change this once we have releases and won't expect people to build from source?

hershen replied:
It's meant to explain how to run the program.
I think building from source will always be an option, so this will always be relevant.
When there'll be releases, we can amend this section.
